File: 5.pl
#!C:/Strawberry/perl/bin/perl.exe

use strict;
use warnings;
use CGI qw(:standard);
use CGI::Carp qw(fatalsToBrowser);
use DBI;

# Get form data
my $nm  = param('name') || '';
my $age = param('age') || 0;

# SQLite connection (creates file if it doesn't exist)
my $db = DBI->connect("dbi:SQLite:dbname=info.db", "", "", {
    RaiseError => 1,
    AutoCommit => 1
}) or die "SQLite connect failed: $DBI::errstr";

# Create table if not exists
$db->do("CREATE TABLE IF NOT EXISTS info (name TEXT, age INTEGER)");

# Insert data
my $insert_q = $db->prepare("INSERT INTO info (name, age) VALUES (?, ?)");
$insert_q->execute($nm, $age);

# Select data
my $select_q = $db->prepare("SELECT name, age FROM info");
$select_q->execute();

# Output HTML
print header();
print start_html("Database Contents");
print "<h2>Current Contents of the 'info' Table</h2>";
print "<table border='1'><tr><th>Name</th><th>Age</th></tr>";

while (my @row = $select_q->fetchrow_array) {
    print "<tr><td>$row[0]</td><td>$row[1]</td></tr>";
}

print "</table>";
print end_html;

# Clean up
$insert_q->finish;
$select_q->finish;
$db->disconnect;

File: 5.php
<?php
// Database configuration
$servername = "localhost";
$username = "root";
$password = ""; // leave empty if no password set
$dbname = "testdb";

// Create connection
$conn = new mysqli($servername, $username, $password, $dbname);

// Check connection
if ($conn->connect_error) {
    die("❌ Connection failed: " . $conn->connect_error);
}

// Insert data if form submitted
$msg = "";
if ($_SERVER["REQUEST_METHOD"] == "POST") {
    $name = trim($conn->real_escape_string($_POST['name']));
    $email = trim($conn->real_escape_string($_POST['email']));

    if (!empty($name) && !empty($email)) {
        $sql = "INSERT INTO users (name, email) VALUES ('$name', '$email')";
        if ($conn->query($sql) === TRUE) {
            $msg = "✅ User added successfully!";
        } else {
            $msg = "❌ Error adding user: " . $conn->error;
        }
    } else {
        $msg = "⚠️ Please enter both name and email.";
    }
}
?>

<!DOCTYPE html>
<html>
<head>
    <title>User Management</title>
    <style>
        body { font-family: Arial; margin: 20px; }
        form, table { margin-bottom: 20px; }
        table { border-collapse: collapse; width: 100%; }
        th, td { padding: 10px; border: 1px solid #ccc; }
        .message { padding: 10px; background: #f0f0f0; margin-bottom: 15px; }
    </style>
</head>
<body>

<h2>Add New User</h2>

<?php if ($msg): ?>
    <div class="message"><?php echo $msg; ?></div>
<?php endif; ?>

<form method="POST" action="">
    <label>Name:</label><br>
    <input type="text" name="name" required><br><br>
    
    <label>Email:</label><br>
    <input type="email" name="email" required><br><br>
    
    <input type="submit" value="Add User">
</form>

<hr>

<h2>Current Users</h2>

<?php
// Fetch and display users
$sql = "SELECT id, name, email FROM users ORDER BY id DESC";
$result = $conn->query($sql);

if ($result && $result->num_rows > 0):
?>
    <table>
        <tr><th>ID</th><th>Name</th><th>Email</th></tr>
        <?php while($row = $result->fetch_assoc()): ?>
        <tr>
            <td><?php echo htmlspecialchars($row["id"]); ?></td>
            <td><?php echo htmlspecialchars($row["name"]); ?></td>
            <td><?php echo htmlspecialchars($row["email"]); ?></td>
        </tr>
        <?php endwhile; ?>
    </table>
<?php else: ?>
    <p>No users found.</p>
<?php endif; ?>

</body>
</html>

<?php $conn->close(); ?>
